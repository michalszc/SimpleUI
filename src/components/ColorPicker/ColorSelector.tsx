import React, { FC } from "react";
import * as CSS from "csstype";
import styled from "styled-components";
import { IStyledTheme } from "../../utils/theme/theme";
import Styles from "../../utils/styles";

export interface ColorSelectorProps {
    color: CSS.DataType.Color;
    shape: 'circle' | 'square' | 'roundedSquare';
    setColor: (value: CSS.DataType.Color) => void;
}

interface IColor 
extends IStyledTheme 
{
    color: CSS.DataType.Color;
    shape: 'circle' | 'square' | 'roundedSquare';
}

const Color = styled.div<IColor>`
    background-color: ${ props => props.color };
    border-radius: ${ ({ shape, theme }) => {
        switch (shape) {
            case 'circle':
                return '50%';
            case 'square':
                return '0';
            case 'roundedSquare':
                return theme.sizes[1];
        }
    }};
    width: ${ props => props.theme.sizes[7] };
    height: ${ props => props.theme.sizes[7] };
    cursor: pointer;
    transition: transform .2s ease-in-out;

    &:hover, &:focus {
        transform: scale(1.15);
    }
`;

const ColorSelector: FC<ColorSelectorProps> = ({
    color, shape, setColor
}) => (
    <Styles>
        <Color 
            color={color}
            shape={shape}
            onClick={() => setColor(color)}
            title={color}
            className={"simpleui-colorpicker-colorselector"}
            data-testid={'test-colorselector'}
        />
    </Styles>
);

export default ColorSelector;
