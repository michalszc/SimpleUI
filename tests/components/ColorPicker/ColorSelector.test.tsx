import React from "react";
import '@testing-library/jest-dom';
import { fireEvent, render } from "@testing-library/react";
import ColorSelector from "../../../src/components/ColorPicker/ColorSelector";

describe("ColorSelector", () => {
    const color = '#FFF';
    const mockFn = jest.fn();
    test("should properly render ColorSelector component - shape circle", () => {
        const container = render(
            <ColorSelector color={color} shape={'circle'} setColor={mockFn} />
        );
        expect(container).toMatchSnapshot();
    });
    test("should properly render ColorSelector component - shape square", () => {
        const container = render(
            <ColorSelector color={color} shape={'square'} setColor={mockFn} />
        );
        expect(container).toMatchSnapshot();
    });
    test("should properly render ColorSelector component - shape roundedSquare", () => {
        const container = render(
            <ColorSelector color={color} shape={'roundedSquare'} setColor={mockFn} />
        );
        expect(container).toMatchSnapshot();
    });
    test("should properly render ColorSelector component with background color", () => {
        const { getByTestId } = render(
            <ColorSelector color={color} shape={'circle'} setColor={mockFn} />
        );
        expect(getByTestId('test-colorselector')).toHaveStyle(`background-color: ${color}`);
    });
    test("should properly trigger setColor function on click", () => {
        const { getByTestId } = render(
            <ColorSelector color={color} shape={'circle'} setColor={mockFn} />
        );
        const colorSelector =  getByTestId('test-colorselector');
        fireEvent.click(colorSelector);
        expect(mockFn).toBeCalled();
    });
});
