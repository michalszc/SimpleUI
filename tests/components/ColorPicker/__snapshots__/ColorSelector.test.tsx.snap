// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorSelector should properly render ColorSelector component - shape circle 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bcXHqe esTXWD simpleui-colorpicker-colorselector"
        color="#FFF"
        data-testid="test-colorselector"
        id="4fzzzxjylrx"
        mode="light"
        shape="circle"
        title="#FFF"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bcXHqe esTXWD simpleui-colorpicker-colorselector"
      color="#FFF"
      data-testid="test-colorselector"
      id="4fzzzxjylrx"
      mode="light"
      shape="circle"
      title="#FFF"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ColorSelector should properly render ColorSelector component - shape roundedSquare 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bcXHqe gSmhYw simpleui-colorpicker-colorselector"
        color="#FFF"
        data-testid="test-colorselector"
        id="4fzzzxjylrx"
        mode="light"
        shape="roundedSquare"
        title="#FFF"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bcXHqe gSmhYw simpleui-colorpicker-colorselector"
      color="#FFF"
      data-testid="test-colorselector"
      id="4fzzzxjylrx"
      mode="light"
      shape="roundedSquare"
      title="#FFF"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ColorSelector should properly render ColorSelector component - shape square 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bcXHqe iYJEST simpleui-colorpicker-colorselector"
        color="#FFF"
        data-testid="test-colorselector"
        id="4fzzzxjylrx"
        mode="light"
        shape="square"
        title="#FFF"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bcXHqe iYJEST simpleui-colorpicker-colorselector"
      color="#FFF"
      data-testid="test-colorselector"
      id="4fzzzxjylrx"
      mode="light"
      shape="square"
      title="#FFF"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
